### ISSUE TITLE
Access to recordings saved on AWS S3 from AntMedia dashboard does not work

### ISSUE THREAD

**Issue by fabiox23** [ant-media/Ant-Media-Server#4776]
### Short description
Recordings saved on AWS S3 cannot be accessed from the AntMedia dashboard. The links to view, embed and download the files do not work. The link to delete the files does work though.

### Environment
Ubuntu 20.04 on EC2
Ant Media Enterprise 2.5.3

### Steps to reproduce
1. Record some streams with S3 enabled
2. Try to access them from the dashboard

**Comment by muratugureminoglu** [comment]
Hi @fabiox23 

You need to change some configurations on the server side. Can you please take a look at the following link[1] ? If you are still having the same issue, please let me know.

[1] https://stackoverflow.com/questions/70683459/vod-not-playing-after-s3-recording-enabled-in-ant-media-server

**Comment by fabiox23** [comment]
Hi, thanks for the comment. That worked well for the play and download options but it is still not working for the embed code option which is probably the most used one.

**Comment by muratugureminoglu** [comment]
Should have worked for embedded code. You can email us (support@antmedia.io) by tagging me and I can check it for you.

**Comment by yashtandon113** [comment]
@fabiox23 
In order to play the recording through Embedded code, you need to create CORS policy on the s3 bucket.
Checkout the following link: https://resources.antmedia.io/docs/s3-integration-and-http-forwarding?highlight=s3#how-to-play-aws-s3-vod-files-with-embedded-web-player

**Comment by fabiox23** [comment]
@mekya Hi, thanks for the information. I have done that but it still doesn't work for me. I have tried the Embedded Player (play.html) with both the filename and the id of the file as the id parameter.

Also, if I upload an mp4 file from the user interface, it is saved on local storage but the play, embedded code and download buttons are still trying to access the file from S3. This is probably an issue in itself.

Thanks.

**Comment by mekya** [comment]
Hi @fabiox23 ,

OK. Then let's confirm the case. I've moved to the backlog to schedule. 
One of my colleague will try to confirm the case.

**Comment by yashtandon113** [comment]
@fabiox23

Today, I tried to reproduce the problem on my end, but everything appears to be fine. Could you please double-check your end? If it still does not work, we can look into it together.

**For the uploaded VOD :-** Yes, having an error is expected because the request is forwarded to the s3 URL. To solve this issue, you can use the user defined script to upload the VOD (uploaded file) to the S3 bucket directly and delete from the local storage. Please see the [Stack Overflow post](https://stackoverflow.com/questions/72585724/how-can-i-automatically-upload-the-vod-file-that-i-am-uploading).

Please use the script mentioned in the answer.

**Comment by fabiox23** [comment]
@yashtandon113

ok, after some testing, I discovered that the embedded code wasn't working due to CORS! On the web server where the embedded code is going to be used, you need to add the header 'Access-Control-Allow-Origin' as a directive in the web server settings.

This CORS directive seems to be missing from the AntMedia own web server running on ports 5080 and 5443. Should it be added?

**Comment by yashtandon113** [comment]
@fabiox23
On the server side, requests are allowed from all origins. You can check the CORS configuration in /webapps/App-name/WEB-INF/web.xml.
It throws CORS issue when request is redirected to the s3 bucket URL so for that you can use Cache Control parameter on the s3 bucket to avoid the CORS issue. You can add by following below steps:

```
> Log in to AWS Management Console
> Go into your S3 bucket
> Select the created streams folder
> Click on "Actions"
> Select "Edit metadata"
> In the "Key" field, select "Cache-Control" from the drop down menu **max-age=10** seconds for Value
> Press "Save" button

```

**Comment by fabiox23** [comment]
Hi, I have tried to edit the metadata on the "streams" folder on AWS. Does this apply to an object (ie the video file) or to the whole folder? Because new videos saved after applying the "Cache-Control" metadata  don't seem to inherit the property. However, the browser seems to be happy with it and is able to play the video... Interestingly, before applying the "Cache-Control" metadata, the video would also play as soon as I opened the "inspect" window on Chrome.
